plugins/instanceof-checkcast/src/main/java/org/qbicc/plugin/instanceofcheckcast/SupersDisplayBuilder.java

```diff
@@ -41,6 +41,7 @@ public void accept(CompilationContext ctxt) {
         tables.assignTypeID(jlo);
         // arrays, including reference array
         CoreClasses coreClasses = CoreClasses.get(ctxt);
+        tables.assignTypeID(coreClasses.getArrayBaseTypeDefinition());
```

Comment:
We have `19` sprinkled around in a few random places as a magic number for typeIds (DispatchTables is guilty).  I think assigning a typeId to the arrayBase class might shift that to 20?

Reply:
Ah yes, I see. I'll edit those accordingly, but we probably should revisit that eventually...

Reply:
Agreed.  It's not a good situation.